CoinDock-api/app/Models/V1/Wallet.php

```diff
@@ -13,14 +20,134 @@ class Wallet extends Model
         'coin_id',
         'user_id',
         'wallet_id',
-        'balance'
+        'balance',
+        'coins',
+        'name'
     ];
 
+    //wallet creation
+    public function WalletCreate($userId, $walletId, $userCoinId, $coins, $balanceInUsd)
+    {
+        Wallet::create([
+            'user_id' => $userId,
+            'wallet_id' => $walletId,
+            'coin_id' => $userCoinId,
+            'coins' => $coins,
+            'balance' => ($balanceInUsd*$coins)
+        ]);
+        return true;
+    }
 
-    public function coin(){
-        return $this->belongsTo(Coin::class);
+    //checking wheather the coin is float or int or scientific
+    public function coinCheck($coin, $coins)
+    {
+        $coinsString = (string)$coins;
+        if (!Str::contains($coinsString, 'E') && Str::contains($coinsString, '.')) {
+            return $coins;
+        } else {
+            return $this->satoshiToCrypt($coin, $coinsString);
+        }
     }
 
+    //converting Sathosis into Corresponding crypto coins
+    public function satoshiToCrypt($coin, $sathosis)
+    {
+        $satsToCoins = array_keys(config('assets.accepted_coins'));
 
-}
+        return ($satsToCoins[$coin]['sats_to_crypt'] * $sathosis);
+    }
+
+    //function to check whether the response is in json or not 
+    public function isJson($string)
+    {
+        json_decode($string);
+        return (json_last_error() == JSON_ERROR_NONE);
+    }
+
+    //getting basePath for for checking Wallet Balance
+    public function basePath($userCoinId, $walletId)
+    {
+        $userCoinName = Coin::whereId($userCoinId)->first()->name;
```

Comment:
Use relation instead of DB call.

if we use  Coin::whereId($userCoinId)->first()->name like,
If the coin table doesn't have any value associated with $userCoinId then we get a null value here.
We call null on the name field. So we get 500 error in the above case.


 so, use Coin::whereId($userCoinId)->first()?->name instead of  Coin::whereId($userCoinId)->first()->name

Reply:
done

Reply:
How we will call relation without saving the Wallet Details. Its in the process of saving the wallet how can we make that walletId is there any alternative?